decode.go

```diff
@@ -134,7 +154,10 @@ func UnmarshalTable(t *ast.Table, v interface{}) (err error) {
 		case *ast.Table:
 			fv, fieldName, found := findField(rv, key)
 			if !found {
-				return fmt.Errorf("line %d: field corresponding to `%s' is not defined in `%T'", av.Line, key, v)
+				if l := getLogger(); l != nil {
```

Comment:
I was thinking the logger was optional so if you don't care about extraneous options that are completely ignored. I could be convinced either way.